src/arrayMethodJoin.js

```diff
@@ -5,7 +5,20 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let string = this[0] === null || typeof this[0]
+    === 'undefined' ? '' : String(this[0]);
```

Comment:
what is going on here? lets rewrite it and make it readable for other developers